p2p/p2p-test-utils/src/lib.rs

```diff
@@ -132,6 +137,41 @@ pub const LONG_TIMEOUT: Duration = Duration::from_secs(600);
 /// A short timeout for events that shouldn't occur.
 pub const SHORT_TIMEOUT: Duration = Duration::from_millis(500);
 
+/// This function can be called on the entire async test's future to ensure that it completes
+/// in a reasonable time.
+///
+/// It can also ensure that the future will complete early once a panic occurs anywhere; this
+/// may be important when debugging sporadically failing tests that can only be debugged by
+/// examining the logs (and running a test for LONG_TIMEOUT time may produce an enormous amount
+/// of logs). Unfortunately, this mechanism will not work properly if multiple tests are failing
+/// simultaneously or if there is a "should_panic" test. So, we only enable it if running under
+/// 'nextest' (which runs each test in a separate process) or if forced via a custom env var.
+pub async fn run_with_timeout<F>(future: F)
+where
+    F: Future,
+{
+    let running_under_nextest = std::env::var("NEXTEST").is_ok();
```

Comment:
I would add a comment + documentation explaining that this is defined by default in nextest.